tfhe/src/integer/server_key/comparator.rs

```diff
@@ -188,6 +178,75 @@ impl<'a> Comparator<'a> {
         self.server_key.key.unchecked_scalar_add_assign(lhs, 1);
     }
 
+    // lhs will be assigned
+    // - 0 if lhs < rhs
+    // - 1 if lhs == rhs
+    // - 2 if lhs > rhs
+    fn scalar_compare_block_assign<PBSOrder: PBSOrderMarker>(
+        &self,
+        lhs: &mut crate::shortint::CiphertextBase<PBSOrder>,
+        scalar: u8,
+    ) {
+        // Same logic as compare_block_assign
+        // but rhs is a scalar
+        let delta =
+            (1u64 << (u64::BITS as u64 - 1)) / (lhs.carry_modulus.0 * lhs.message_modulus.0) as u64;
+        let plaintext = Plaintext((scalar as u64) * delta);
+        crate::core_crypto::algorithms::lwe_ciphertext_plaintext_sub_assign(&mut lhs.ct, plaintext);
+        self.server_key
+            .key
+            .apply_lookup_table_assign(lhs, &self.sign_accumulator);
+
+        self.server_key.key.unchecked_scalar_add_assign(lhs, 1);
+    }
+
+    fn reduce_two_sign_blocks_assign<PBSOrder: PBSOrderMarker>(
+        &self,
+        msb_sign: &mut crate::shortint::CiphertextBase<PBSOrder>,
+        lsb_sign: &crate::shortint::CiphertextBase<PBSOrder>,
+    ) {
+        // We don't use pack_block_assign as the offset '4' does not depend on params
+        self.server_key.key.unchecked_scalar_mul_assign(msb_sign, 4);
+        self.server_key.key.unchecked_add_assign(msb_sign, lsb_sign);
+
+        self.server_key
+            .key
+            .apply_lookup_table_assign(msb_sign, &self.selection_accumulator);
+    }
+
+    /// Reduces a vec containing shortint blocks that encrypts a sign
+    /// (inferior, equal, superior) to one single shortint block containing the
+    /// final sign
+    fn reduce_signs_parallelized<PBSOrder>(
+        &self,
+        mut sign_blocks: Vec<crate::shortint::CiphertextBase<PBSOrder>>,
+    ) -> crate::shortint::CiphertextBase<PBSOrder>
+    where
+        PBSOrder: PBSOrderMarker,
+    {
+        let mut sign_blocks_2 = Vec::with_capacity(sign_blocks.len() / 2);
+        while sign_blocks.len() != 1 {
+            sign_blocks
+                .par_chunks_exact(2)
+                .map(|chunk| {
+                    let (low, high) = (&chunk[0], &chunk[1]);
+                    let mut high = high.clone();
+                    self.reduce_two_sign_blocks_assign(&mut high, low);
+                    high
+                })
+                .collect_into_vec(&mut sign_blocks_2);
+
+            if (sign_blocks.len() % 2) == 1 {
+                sign_blocks_2.push(sign_blocks[sign_blocks.len() - 1].clone());
+            }
+
+            std::mem::swap(&mut sign_blocks_2, &mut sign_blocks);
+        }
+        let selection = sign_blocks.drain(..).next().unwrap();
```

Comment:
I think into_iter().next().unwrap() is the better thing because in our case we want to move the first element out of the vec (and the vec only has one element), we don't need the vec after since its going to be empty. Drain moves from the vec without moving (taking ownership) of the vec itself.

In this case its a very minor difference that